ic-canister/ic-canister-macros/src/derive.rs

```diff
@@ -103,8 +103,8 @@ pub fn derive_canister(input: TokenStream) -> TokenStream {
                 .push(quote! {#field_name : <#field_type as ::ic_storage::IcStorage>::get()});
             state_fields_test.push(quote! {#field_name : ::std::rc::Rc::new(::std::cell::RefCell::new(<#field_type as ::std::default::Default>::default()))});
 
-            if is_stable {
-                stable_fields.push((field_name, field_type));
+            if is_stable && stable_field.is_none() {
```

Comment:
Doesn't ths mean that if the `State` struct has more then one stable field, the macro will just disregard all except the first one. I think we want to throw a compile error in this case.

Reply:
Fixed